tests/bluechi_test/machine.py

```diff
@@ -276,10 +276,22 @@ def gather_coverage(self, data_coverage_dir: str) -> None:
         coverage_file = f"{BluechiMachine.gcda_file_location}/coverage-{self.name}.info"
 
         LOGGER.info(f"Generating info file '{coverage_file}' started")
-        bluechi_version = get_env_value("BLUECHI_VERSION", "unknown")
+
+        bluechi_build_dir = ".*/rpmbuild/BUILD/bluechi[^/]*"
+
+        # Check if src directory exists, if not run copy-src-files.sh
+        result, output = self.client.exec_run("test -d /var/tmp/bluechi-coverage/src")
+        if result != 0:
+            LOGGER.info("src directory doesn't exist, running copy-src-files.sh")
+            result, output = self.client.exec_run(
+                "/usr/share/bluechi-coverage/bin/setup-src-dir-for-coverage.sh"
+            )
+            LOGGER.info(f"copy-src-files.sh result: {result}, output: {output}")
+        else:
+            LOGGER.info("src directory already exists")
```

Comment:
I'd probably remove the else path here. The log doesn't provide additional value since the execution of `test -d ...` is generating similar logs already.